AirCasting/SessionViews/SessionCartView.swift

```diff
@@ -222,18 +214,56 @@ private extension SessionCartView {
                         .frame(height: 120)
                         .disabled(true)
                     HStack() {
-                            Text(startTime)
+                            startTime
                             Spacer()
                         Text("\(Strings.SessionCartView.avgSession) \(selectedStream.unitSymbol ?? "")")
                             Spacer()
-                            Text(endTime)
+                            endTime
                     }.foregroundColor(.aircastingGray)
                     .font(Font.muli(size: 13, weight: .semibold))
                 }
             }
         }
     }
     
+    var startTime: some View {
+        let formatter = Constants.dataFormatter
+        
+        if !(session.isMobile && session.isActive && session.deviceType == .MIC) {
+            formatter.timeZone = TimeZone.init(abbreviation: "UTC")
+        }
+            
+            guard var start = session.startTime else { return Text("") }
+     
+        if session.isFixed && session.measurementStreams == [] {
+            start = start.currentUTCTimeZoneDate
+        }
+        
+        let string = formatter.string(from: start)
+        return Text(string)
+        }
+    
+    var endTime: some View {
+        let formatter = Constants.dataFormatter
+        
+        if !(session.isMobile && session.isActive && session.deviceType == .MIC) {
+            formatter.timeZone =  TimeZone.init(abbreviation: "UTC")
+        }
+            
+            var end = session.endTime ?? Date()
+        
+        if session.isMobile && session.deviceType == .AIRBEAM3 && session.endTime == nil {
```

Comment:
Please, test on dormants on Michael account, bc sometimes `deviceType` is nil even though it supposed to be Airbeam